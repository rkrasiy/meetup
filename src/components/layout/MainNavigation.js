import { ALL_MEETUP_PAGE, FAVORITES_PAGE, NEW_MEETUP_PAGE } from "./../../utils/constants";
import { useEffect, useState } from "react";

import classes from "./MainNavigation.module.css";

export default function MainNavigation({ setPage }) {
  const [scrolling, setScrolling] = useState(false);
  const [scrollTop, setScrollTop] = useState(0);

  useEffect(() => {
    const onScroll = e => {
      setScrollTop(e.target.documentElement.scrollTop);
      setScrolling(e.target.documentElement.scrollTop > scrollTop);
    };
    window.addEventListener("scroll", onScroll);

    return () => window.removeEventListener("scroll", onScroll);
  }, [scrollTop]);
console.log(scrolling)
  return (
    <header className={`${classes.header} ${!scrolling ? classes.sticky : ''}`} data-test="navigation-header">
      <div className={classes.logo}>React Meetups</div>
      <nav>
        <ul>
          <li>
            <a href="#" onClick={() => setPage(ALL_MEETUP_PAGE)}>
              All Meetups
            </a>
          </li>

          <li>
            <a href="#" onClick={() => setPage(NEW_MEETUP_PAGE)}>
              Add New Meetup
            </a>
          </li>
          <li>
            <a href="#" onClick={() => setPage(FAVORITES_PAGE)}>
              My Favorites
              <span className={classes.badge}>{0}</span>
            </a>
          </li>
        </ul>
      </nav>
    </header>
  );
}
