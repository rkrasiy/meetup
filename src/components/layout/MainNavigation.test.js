/* eslint-disable testing-library/no-debugging-utils */
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import MainNavigation from "./MainNavigation"

test("<MainNavigation/> renders without crashing", () => {
  const wrapper = shallow(<MainNavigation />);
  expect(wrapper.exists()).toBe(true);
});

describe('Check component MainNavigation', ()=>{
  let wrapper = shallow(<MainNavigation />);

  beforeEach(()=> {
    wrapper = shallow(<MainNavigation />);
  })

  test('must be shown <MainNavigation /> correctly', ()=> {
    expect(wrapper).toMatchSnapshot();
  })

  test('must shown 0', ()=> {
     const wrapper = shallow(<MainNavigation />);
      const counterText = wrapper.find('span').text().trim();
      expect(counterText).toBe('0');
  })

  test('must shown 3', ()=> {
     const wrapper = shallow(<MainNavigation  favoriteCount={3} />);
      const counterText = wrapper.find('span').text().trim();
      expect(counterText).toBe('3');
  })
})