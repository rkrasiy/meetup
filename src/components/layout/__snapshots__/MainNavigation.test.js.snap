// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check component MainNavigation must be shown <MainNavigation /> correctly 1`] = `ShallowWrapper {}`;
